tests_e2e/tests/ext_policy/ext_policy.py

```diff
@@ -0,0 +1,233 @@
+
+        # Prepare no-config, single-config, and multi-config extension to test. Extensions with settings and extensions
+        # without settings have different status reporting logic, so we should test all cases.
+        unique = str(uuid.uuid4())
+        test_file = f"waagent-test.{unique}"
+
+        # CustomScript is a single-config extension.
+        custom_script = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.CustomScript,
+                                          resource_name="CustomScript"),
+            {'commandToExecute': f"echo '{unique}' > /tmp/{test_file}"}
+        )
+
+        # RunCommandHandler is a multi-config extension, so we set up two instances (configurations) here and test both.
+        run_command = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.RunCommandHandler,
+                                          resource_name="RunCommandHandler"),
+            {'source': {'script': f"echo '{unique}' > /tmp/{test_file}"}}
+        )
+        unique2 = str(uuid.uuid4())
+        test_file2 = f"waagent-test.{unique2}"
+        run_command_2 = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.RunCommandHandler,
+                                          resource_name="RunCommandHandler2"),
+            {'source': {'script': f"echo '{unique2}' > /tmp/{test_file2}"}}
+        )
+
+        # AzureMonitorLinuxAgent is a no-config extension (extension without settings).
+        azure_monitor = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.AzureMonitorLinuxAgent,
+                                          resource_name="AzureMonitorLinuxAgent"),
+            None
+        )
+
+        # Enable policy via conf
+        log.info("Enabling policy via conf file on the test VM [%s]", self._context.vm.name)
+        self._ssh_client.run_command("update-waagent-conf Debug.EnableExtensionPolicy=y", use_sudo=True)
+
+        # Only allowlisted extensions should be processed.
+        # We only allowlist CustomScript: CustomScript should be enabled, RunCommand and AzureMonitor should fail.
+        # (Note that CustomScript blocked by policy is tested in a later test case.)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": False,
+                    "extensions": {
+                        "Microsoft.Azure.Extensions.CustomScript": {}
+                    }
+                }
+            }
+        self._create_policy_file(policy)
+        self._operation_should_succeed("enable", custom_script)
+        self._operation_should_fail("enable", run_command)
+        if VmExtensionIds.AzureMonitorLinuxAgent.supports_distro((self._ssh_client.run_command("get_distro.py").rstrip())):
```

Comment:
how much coverage are we getting for this case?

Reply:
I've changed ext_policy.yml to run on all endorsed distros, so this case will be run on all distros that support AzureMonitorLinuxAgent.

Reply:
yes, no need for that check